internal/pkg/cli/deploy/workload.go

```diff
@@ -302,27 +314,62 @@ func (d *workloadDeployer) forceDeploy(in *forceDeployInput) error {
 	return nil
 }
 
-func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, error) {
+func (d *workloadDeployer) uploadContainerImage(imgBuilderPusher imageBuilderPusher) (*string, map[string]string, error) {
 	required, err := manifest.DockerfileBuildRequired(d.mft)
 	if err != nil {
-		return nil, err
-	}
-	if !required {
-		return nil, nil
+		return nil, nil, err
 	}
-	// If it is built from local Dockerfile, build and push to the ECR repo.
-	buildArg, err := buildArgs(d.name, d.imageTag, d.workspacePath, d.mft)
+	UUIDTag, err := generateUUID()
 	if err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	var digest string
+	if required {
+		// If it is built from local Dockerfile, build and push to the ECR repo.
+		buildArg, err := buildArgs(d.name, d.imageTag, d.gitTag, UUIDTag, d.workspacePath, d.mft)
+		if err != nil {
+			return nil, nil, err
+		}
+		digest, err = imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), buildArg)
+		if err != nil {
+			return nil, nil, fmt.Errorf("build and push main container image: %w", err)
+		}
 	}
-	digest, err := imgBuilderPusher.BuildAndPush(dockerengine.New(exec.NewCmd()), buildArg)
+	args := scBuildArgs(d.name, d.gitTag, UUIDTag, d.workspacePath, d.mft)
+	scDigests := make(map[string]string, len(args))
+	ctx := context.Background()
+	g, _ := errgroup.WithContext(ctx)
+	for k, v := range args {
+		scName := k
+		dArgs := v
```

Comment:
Can we include a comment to https://go.dev/doc/effective_go#goroutines to explain why it's done this way:
```suggestion
	for name, imgCfg := range imageConfigs {
		name := name // See https://go.dev/doc/effective_go#goroutines 
		imgCfg := imgCfg
```